### PR TITLE
Follow up to #2331: Refactoring backend controller "Category" & testcase

### PR DESCRIPTION
**Pull request by pascalheidmann**
### 1. Why is this change necessary?
Code quality & translation 

### 2. What does this change do, exactly?
Building upon #2331 , adding:
- Introduce translation to `Shopware_Controllers_Backend_Category` and its test case
- Refactor both the controller and its test case

### 4. Please link to the relevant issues (if any).
This is a follow up to #2331 

### 6. Checklist

- [x] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [x] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: engine/Shopware/Controllers/Backend/Category.php [hunk 1]
```diff
@@ -42,7 +50,25 @@ class Shopware_Controllers_Backend_Category extends Shopware_Controllers_Backend
...
+     * @var ModelManager
+     */
+    private $em;
+
+    /**
+     * @var Shopware_Components_Snippet_Manager
+     */
+    private $translation;
+
+    public function __construct(ModelManager $em, Shopware_Components_Snippet_Manager $translation)
```
Comment (shyim) [inline engine/Shopware/Controllers/Backend/Category.php]: Can you alias the SnippetManager?

Comment (pascalheidmann) [reply engine/Shopware/Controllers/Backend/Category.php]: I would prefer introducing an interface and using that instead but that would blow up this ticket's intention ;)

Comment (shyim) [reply engine/Shopware/Controllers/Backend/Category.php]: Maybe we can alias it in 5.7 branch. So both would be possible :)

File: snippets/backend/category/main.ini [hunk 2]
```diff
@@ -136,6 +137,7 @@ category/action/add/success = "[0]x Artikel Zuordnungen gespeichert"
 category/action/remove/failure = "Während des Löschens der Artikel ist folgender Fehler aufgetreten:"
 category/action/remove/success = "[0]x Artikel Zuordnungen gelöscht"
 main/window/button/save = "Speichern"
+saveDetailInvalidCategoryId = "Ungültige CategoryId"
```
Comment (mitelg) [inline snippets/backend/category/main.ini]:
```suggestion
saveDetailInvalidCategoryId = "Ungültige Kategorie-ID"
```

**Comment by shyim** [comment]
Thanks for your Contribution! :blue_heart:

**Pull request closed without merging**